CMakeLists.txt

```diff
@@ -1,48 +1,85 @@
-# Provide general project configuration setup
-#
-cmake_minimum_required(VERSION 3.21)
+cmake_minimum_required(VERSION 3.19)
+project(StatusDesktop)
 
-set(STATUS_QT_VERSION 6.3)
-set(STATUS_VERSION 0.6.0)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-project(status-desktop
-  VERSION ${STATUS_VERSION}
-  LANGUAGES CXX)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
-set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
+if (APPLE)
+    set(MACOS_VERSION_MIN_FLAGS -mmacosx-version-min=11.0)
+endif()
 
-set(PROJECT_ORGANIZATION_DOMAIN "status.im")
-set(PROJECT_ORGANIZATION_NAME "Status")
-set(PROJECT_APPLICATION_NAME "Status Desktop")
+set(STATUSQ_SHADOW_BUILD ON)
+set(STATUSQ_BUILD_SANDBOX OFF)
+set(STATUSQ_BUILD_SANITY_CHECKER OFF)
+set(STATUSQ_BUILD_TESTS OFF)
 
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+find_package(QT NAMES Qt5 Qt6 REQUIRED COMPONENTS Core)
+find_package(OpenSSL REQUIRED)
 
-include(CTest)
-enable_testing()
+find_package(Qt${QT_VERSION_MAJOR} COMPONENTS
+    Core Gui Quick QuickControls2 Widgets Test Multimedia WebChannel Svg
+    QuickTest Qml WebView REQUIRED)
 
-if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm64" AND APPLE)
-  set(STATUSGO_FORCE_ARCH ${CMAKE_HOST_SYSTEM_PROCESSOR})
-endif()
+add_subdirectory(ui/StatusQ)
+
+add_executable(
+  ${PROJECT_NAME}
+  main.cpp
+)
+
+set(VENDOR_DIR "${CMAKE_SOURCE_DIR}/vendor")
+
+add_custom_target(BuildStatusDekstopLib ALL
+  BYPRODUCTS
+    ${CMAKE_SOURCE_DIR}/bin/libnim_status_client.a
+    ${VENDOR_DIR}/DOtherSide/build/lib/libDOtherSideStatic.a
+    ${VENDOR_DIR}/QR-Code-generator/c/libqrcodegen.a
+    ${VENDOR_DIR}/status-go/build/bin/libstatus.so
+    ${VENDOR_DIR}/status-keycard-go/build/libkeycard/libkeycard.so
+  COMMAND make nim_status_lib
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  COMMENT "Building nim_status_lib"
+)
+
+add_library(DOtherSideStatic STATIC IMPORTED)
+set_target_properties(DOtherSideStatic PROPERTIES IMPORTED_LOCATION
+    "${VENDOR_DIR}/DOtherSide/build/lib/libDOtherSideStatic.a")
```

Comment:
`CMAKE_STATIC_LIBRARY_SUFFIX` makes it more portable :) The suffix is different on various OSs

Reply:
Oh yea, `CMAKE_STATIC_LIBRARY_PREFIX`